src/bridge/cockpitrouter.c

```diff
@@ -323,6 +439,141 @@ process_open_peer (CockpitRouter *self,
   return cockpit_peer_handle (peer, channel, options, data);
 }
 
+static GBytes *
+substitute_json_string (const gchar *variable,
+                        gpointer user_data)
+{
+  const gchar *value;
+  JsonObject *options = user_data;
+  if (options && cockpit_json_get_string (options, variable, "", &value))
+    return g_bytes_new (value, strlen (value));
+  else if (options)
+    g_message ("Couldn't get argument for bridge: got invalid value for '%s'", variable);
+
+  return g_bytes_new ("", 0);
+}
+
+static JsonArray *
+split_json_array (JsonArray *arr,
+                  gint start,
+                  gint end)
+{
+  JsonArray *new;
+  gint length;
+  gint i;
+
+  g_return_val_if_fail (arr != NULL, NULL);
+
+  new = json_array_new ();
+  length = json_array_get_length (arr);
+  if (end > length)
+    end = length;
+
+  for (i = start; i < end; i++)
+    json_array_add_string_element (new, json_array_get_string_element (arr, i));
+
+  return new;
+}
+
+static void
+add_dynamic_args_to_array (JsonArray *arr,
+                           gchar **config_args,
+                           JsonObject *options)
+{
+  gint length;
+  gint i;
+
+  g_return_if_fail (config_args != NULL);
+  g_return_if_fail (arr != NULL);
+
+  length = g_strv_length (config_args);
+  for (i = 0; i < length; i++)
+    {
+      gchar *final = NULL;
+      GByteArray *array = g_byte_array_new ();
+      GBytes *input = g_bytes_new_static (config_args[i], strlen(config_args[i]) + 1);
+      GList *output = cockpit_template_expand (input, substitute_json_string,
+                                               "${", "}", options);
+      GList *l;
+      for (l = output; l != NULL; l = g_list_next (l))
+        {
+          gsize size;
+          gconstpointer data = g_bytes_get_data (l->data, &size);
+          g_byte_array_append (array, data, size);
+        }
+
+      final = (gchar *) g_byte_array_free (array, FALSE);
```

Comment:
What sets the null character here? How do we turn the above array into a string? Missing a ```g_byte_array_append``` with a "\0" char?